Parsers/ASimRegistryEvent/Parsers/vimRegistryEventMicrosoftWindowsEvent.yaml

```diff
@@ -13,84 +13,141 @@ References:
-                          ActingProcessId = tostring(toint(tostring(EventData.ProcessId))), 
-                          ActingProcessName = tostring(EventData.ProcessName)
-                    | extend     
-                    // -- Aliases
-                          User = ActorUsername,
-                          UserId = ActorUserId,
-                          Dvc = DvcHostname,
-                          Process = ActingProcessName
-                      }; RegistryEvents
\ No newline at end of file
+    let RegistryEvents=(){
+        let ASIM_GetWindowsUserType = (username:string, sid:string) { 
+            case ( 
+                username endswith "$" or sid in ("S-1-5-18", "S-1-5-19", "S-1-5-20"), "Machine"
+                , isempty(username) or sid == "S-1-0-0", ""
+                , "Regular"
+            )
+        };
+        let ASIM_GetAccountType = (sid:string) { 
+            iif ( 
+                sid in ("S-1-0-0","S-1-5-18", "S-1-5-19", "S-1-5-20"), "Simple"
+                , "Windows"
+            )
+        };
+        let ASIM_ParseWindowsEvents = (WindowsEvent:(EventData:dynamic)) {
+            WindowsEvent
+            | extend ActorUsername = strcat(EventData.SubjectDomainName, @'\', EventData.SubjectUserName)
+                , ActorDomainName = tostring(EventData.SubjectDomainName)
+                , ActorUserId = tostring(EventData.SubjectUserSid)
+                , ActorSessionId = tostring(EventData.SubjectLogonId)
+                , ActingProcessName = tostring(EventData.ProcessName)
+                , ActingProcessId = tostring(toint(tolong(EventData.ProcessId)))
+                , RegistryKey = iif(EventData.ObjectName startswith @"\REGISTRY\MACHINE", replace_string(tostring(EventData.ObjectName), @"\REGISTRY\MACHINE","HKEY_LOCAL_MACHINE")
+                , replace_string(tostring(EventData.ObjectName), @"\REGISTRY\USER","HKEY_USERS"))
+                , Type = "WindowsEvent"
+        };
+        let ASIM_ParseSecurityEvents = (SecurityEvent:(SubjectDomainName:string,SubjectUserName:string,ProcessId:string,ObjectName:string,SubjectUserSid:string,SubjectLogonId:string,ProcessName:string)) {
+            SecurityEvent
+            | extend ActorUsername = strcat(SubjectDomainName, @'\', SubjectUserName)
+                , ActorDomainName = SubjectDomainName
+                , ActingProcessId = tostring(toint(tolong(ProcessId)))
+                , RegistryKey = iif(ObjectName startswith @"\REGISTRY\MACHINE", replace_string(ObjectName, @"\REGISTRY\MACHINE","HKEY_LOCAL_MACHINE")
+                , replace_string(ObjectName, @"\REGISTRY\USER","HKEY_USERS"))
+                , Type = "SecurityEvent"
+            | project-rename ActorUserId = SubjectUserSid
+                , ActorSessionId = SubjectLogonId
+                , ActingProcessName = ProcessName
+        };
+        let EventTypeLookup = datatable (AccessMask:string,EventType:string)
+        [
+            "0x1", "RegistryValueRead"
```

Comment:
I need to add those to the doc and to the validation? correct? Actually, you can update the validation CSV as part of this PR.

Which make me ask: did you run tests?

Reply:
Done